hr_evaluation_360/models/hr_evaluation_plan_phase.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Luis Felipe Mileo - <mileo@kmee.com.br> - KMEE
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from dateutil import parser
+import time
+
+from openerp import api, fields, models
+
+evaluation_360_list = []
+
+
+def evaluation_360(evaluation_func):
+    evaluation_360_list.append(evaluation_func)
+    return evaluation_360
+
+
+class HrEvaluationPlanPhase(models.Model):
+
+    _inherit = 'hr_evaluation.plan.phase'
+
+    action = fields.Selection(
+        selection_add=[
+            ('360', u'360-degree feedback'),
+            ('360-anonymous', u'360-degree anonymous feedback'),
+        ]
+    )
+
+
+class HrEvaluationEvaluation(models.Model):
+
+    _inherit = 'hr_evaluation.evaluation'
+
+    @evaluation_360
+    def _get_360_evaluation_child(self, evaluation):
+        return evaluation.employee_id.child_ids
+
+    @evaluation_360
+    def _get_360_evaluation_parent(self, evaluation):
+        return evaluation.employee_id.parent_id
+
+    @evaluation_360
+    def _get_360_evaluation_myself(self, evaluation):
+        return evaluation.employee_id
+
+    @evaluation_360
+    def _get_360_evaluation_department(self, evaluation):
+        return self.env['hr.employee'].search(
+            [('department_id', '=',
+              evaluation.employee_id.department_id.id)]
+        )
+
+    @api.multi
+    def button_plan_in_progress(self):
+        hr_eval_inter_obj = self.env['hr.evaluation.interview']
+        for evaluation in self:
+            wait = False
+            for phase in evaluation.plan_id.phase_ids:
+                if phase.action not in ('360', '360-anonymous'):
+                    return super(HrEvaluationEvaluation,
+                                 self).button_plan_in_progress()
+
+                children = self.env['hr.employee']
+                for item in evaluation_360_list:
+                    children |= item(self, evaluation)
```

Comment:
In a 360 evaluation not every evaluator is an Employee.
AFAIK it is common to ask evaluations from other Partners, such as Customers and Suppliers.